mmf/modules/encoders.py

```diff
@@ -230,6 +241,50 @@ def forward(self, x):
         return out  # BxNx2048
 
 
+@registry.register_encoder("resnet")
+class ResNetEncoder(Encoder):
```

Comment:
@kdexd 's PR #336 does have a more generic version of this. Not sure what his current plans are regarding landing this.

Reply:
#336 has some merge conflicts, and I am still revamping it with new upstream changes. 
However, image encoder seems like an important feature that would be nice to have in upstream ASAP.

I scoped it out separately and made a quick short PR: https://github.com/facebookresearch/mmf/pull/728
Once this is merged, it could be used in this PR if serves the purpose (I will re-use it in my PR).

Reply:
I have removed the support for torchvision resnets in this method and made it specific to detectron2 resnet backbones only. @kdexd's implementation won't overlap with this anymore.